code/game/objects/effects/spawners/mobspawner.dm

```diff
@@ -0,0 +1,75 @@
+/obj/effect/spawner/lootdrop/randomthreat
+	var/static/mob_category = rand(1, 3)
+
+/obj/effect/spawner/lootdrop/randomthreat/Initialize(mapload)
+	switch(mob_category)
+		if(1)
+			loot = list(
+				/obj/effect/spawner/lootdrop/hivebotspawner
+			)
+		if(2)
+			loot = list(
+				/obj/effect/spawner/lootdrop/carpspawner
+			)
+		if(3)
+			loot = list(
+				/obj/effect/spawner/lootdrop/spiderspawner
+			)
+	. = ..()
+
+/obj/effect/spawner/lootdrop/xenospawner
+	name = "Improper xenomorph spawner"
```

Comment:
what are you doing with the names?

Reply:
Sometimes, I don't even know what I'm doing.